packages/client/src/utils/locationUtils.ts

```diff
@@ -275,3 +275,30 @@ export function getLocationHierarchy(
     [camelCasedJurisdictionType(jurisdictionType)]: id
   })
 }
+
+export function generateFullAddress(
+  address: Address,
+  offlineData: IOfflineData
+): (string | null | undefined)[] {
+  const district =
+    address.district && offlineData.locations[address.district].name
+
+  const state = address.state && offlineData.locations[address.state].name
+
+  const eventLocationLevel3 =
+    address?.line?.[10] && offlineData.locations[address.line[10]]?.name
+
+  const eventLocationLevel4 =
+    address?.line?.[11] && offlineData.locations[address.line[11]]?.name
+
+  const eventLocationLevel5 =
+    address?.line?.[12] && offlineData.locations[address.line[12]]?.name
+
+  return [
+    eventLocationLevel5,
+    eventLocationLevel4,
+    eventLocationLevel3,
+    district,
+    state
+  ]
+}
```

Comment:
I would say having the function signature as
```
export function generateFullAddress(
  address: Address,
  offlineData: IOfflineData
): string[]
```
would be more preferable. We can achieve that by doing it like this: 
```
return [
    eventLocationLevel5,
    eventLocationLevel4,
    eventLocationLevel3,
    district,
    state
  ].filter(Boolean)
```